pkg/filter/topicmapper/topic.go

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package topicmapper
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/megaease/easegress/pkg/logger"
+)
+
+type topicMapFunc func(mqttTopic string) (topic string, headers map[string]string, err error)
+
+func getPolicyRoute(routes []*PolicyRe) map[string]*regexp.Regexp {
+	ans := make(map[string]*regexp.Regexp)
+	for _, route := range routes {
+		r, err := regexp.Compile(route.MatchExpr)
+		if err != nil {
+			logger.SpanErrorf(nil, "topicMapper policy <%s> match expr <%s> compile failed: %v", route.Name, route.MatchExpr, err)
+		} else {
+			ans[route.Name] = r
+		}
+	}
+	return ans
+}
+
+type topicRouteType map[string][]*regexp.Regexp
+
+func getTopicRoute(p *Policy) topicRouteType {
+	m := make(map[string][]*regexp.Regexp)
+	for _, route := range p.Route {
+		for _, expr := range route.Exprs {
+			r, err := regexp.Compile(expr)
+			if err != nil {
+				logger.SpanErrorf(nil, "topicMapper policy <%s> topic route expr <%s> compile failed: %v", p.Name, expr, err)
+			} else {
+				m[route.Topic] = append(m[route.Topic], r)
+			}
+		}
+	}
+	return m
+}
+
+func getPolicyMap(ps []*Policy) map[string]*Policy {
+	ans := make(map[string]*Policy)
```

Comment:
what does `ans` mean?

Reply:
i usually use `ans`, `res` to represent result or answer of short functions, i will update this.

Reply:
Thanks, please use `res` as most developers do to improve readability.

Reply:
yeah, `ans` is a habit from LeetCode... i will update that